gridcomps/History/MAPL_HistoryGridComp.F90

```diff
@@ -5369,27 +5388,29 @@ subroutine regen_rcx_for_obs_platform (config, nlist, list, rc)
        i=index(line, 'PLATFORM.')
        j=index(line, ':')
        marker=line(1:j)
-       call scan_begin(unitr, marker, .true.)     
+       call scan_begin(unitr, marker, .true.)
        call scan_contain(unitr, 'geovals_fields:', .false.)
        ios=0
        ngeoval=0
+       nseg_ub=0
        do while (ios == 0)
           read (unitr, '(A)' ) line
-          i=index(line, '::')
-          if (i==0) then
+          con = .not.(adjustl(trim(line))=='::')
+          if (con) then
              ngeoval = ngeoval + 1
              call  split_string_by_space (line, length_mx, mxseg, &
                   nseg, str_piece, status)
+             nseg_ub = max(nseg_ub, nseg)
           else
              exit
           endif
```

Comment:
Avoid `else` - makes for better code.
```suggestion
          if (.not. con) exit
          ngeoval = ngeoval + 1
          call  split_string_by_space (line, length_mx, mxseg, &
               nseg, str_piece, status)
          nseg_ub = max(nseg_ub, nseg)
```

Reply:
Thanks.  This change is added into another PR.